src/LowerTriangularArrays/lower_triangular_array.jl

```diff
@@ -617,27 +670,28 @@ find_L(a::LowerTriangularArray, rest) = a
 find_L(::Any, rest) = find_L(rest)
 
 function Base.similar(
-    bc::Broadcasted{LowerTriangularStyle{N, ArrayType}},
+    bc::Broadcasted{LowerTriangularStyle{N, ArrayType, S}},
     ::Type{T},
-) where {N, ArrayType, T}
+) where {N, ArrayType, S, T}
     L = find_L(bc)
-    return LowerTriangularArray{T, N, ArrayType{T,N}}(undef, size(L; as=Matrix))
+    return LowerTriangularArray{T, N, ArrayType{T,N}, S}(undef, size(L; as=Matrix))
```

Comment:
same here, I think the broadcasted spectrum should be reused to construct a new lowertriangular array

Reply:
For RingGrids I solved this like
```julia
function Base.similar(bc::Broadcasted{FieldGPUStyle{N, ArrayType, Grid}}, ::Type{T}) where {N, ArrayType, Grid, T}
    for maybe_field in bc.args
        if maybe_field isa AbstractField
            return similar(maybe_field, T)
        end
    end
end
```

So this looks for the first field in broadcasted and reuses that `grid`. This is necessary because you can write `1 .+ field` or `field .+ 1` or `field1 .+ field2`. So which `grid` to use in either case? I thought it would be an easily rule to use the first `grid` respectively

Reply:
For in-place operations like `L .= L1 .+ L2 .+ 1` I don't think this applies as each `L, L1, L2` retain their respective `spectrum`